File: Problem_Solving/Letcode_Problem_Solutions_with_python/461_hamming_distance.py
class Solution:
    def hammingDistance(self, x: int, y: int) -> int:
        cnt = 0
        while x or y:
            print(x, bin(x), y, bin(y))
            if x & 1 ^ y & 1:
                cnt += 1
            x >>= 1
            y >>= 1
        return cnt


if __name__ == "__main__":
    ans = Solution().hammingDistance(5,4)
    print(ans)

File: Problem_Solving/Leet_code_problem_solutions_with_GoLang/461_hamming_distance.go
package main

import "fmt"

func hammingDistance(x int, y int) int {

	cnt := 0
	for x != 0 || y != 0 {
		if (x&1 ^ y&1) == 1 {
			cnt += 1
		}
		x = x >> 1
		y = y >> 1
	}
	return cnt
}

func main() {
	ans := hammingDistance(5, 4)
	fmt.Println(ans)
}
